public/components/FeatureAnywhereContextMenu/AssociatedDetectors/containers/AssociatedDetectors.tsx

```diff
@@ -0,0 +1,347 @@
+            resp,
+            'hits',
+            []
+          );
+          const curSelectedDetectors = getAssociatedDetectors(
+            Object.values(allDetectors),
+            savedAugmentObjectsArr
+          );
+          setSelectedDetectors(curSelectedDetectors);
+          setIsLoadingFinalDetectors(false);
+        }
+      })
+      .catch((error) => {
+        core.notifications.toasts.addDanger(
+          prettifyErrorMessage(`Unable to fetch associated detectors: ${error}`)
+        );
+      });
+  }, [allDetectors]);
+
+  // cross checks all the detectors that exist with all the savedAugment Objects to only display ones
+  // that are associated to the current visualization
+  const getAssociatedDetectors = (
+    detectors: DetectorListItem[],
+    savedAugmentObjects: ISavedAugmentVis[]
+  ) => {
+    // Filter all savedAugmentObjects that aren't linked to the specific visualization
+    const savedAugmentForThisVisualization: ISavedAugmentVis[] =
+      savedAugmentObjects.filter(
+        (savedObj) => get(savedObj, 'visId', '') === embeddable.vis.id
+      );
+
+    // Map all detector IDs for all the found augmented vis objects
+    const savedAugmentDetectorsSet = new Set(
+      savedAugmentForThisVisualization.map((savedObject) =>
+        get(savedObject, 'pluginResourceId', '')
+      )
+    );
+
+    // filter out any detectors that aren't on the set of detectors IDs from the augmented vis objects.
+    const detectorsToDisplay = detectors.filter((detector) =>
+      savedAugmentDetectorsSet.has(detector.id)
+    );
+    return detectorsToDisplay;
+  };
+
+  const onUnlinkDetector = async () => {
+    setIsLoadingFinalDetectors(true);
+    await savedObjectLoader.findAll().then(async (resp: any) => {
+      if (resp != undefined) {
+        const savedAugmentObjects: ISavedAugmentVis[] = get(resp, 'hits', []);
+        // gets all the saved object for this visualization
+        const savedAugmentForThisVisualization: ISavedAugmentVis[] =
+          savedAugmentObjects.filter(
+            (savedObj) => get(savedObj, 'visId', '') === embeddable.vis.id
+          );
+
+        // find saved Augment object matching detector we want to unlink
+        // There should only be one detector and vis pairing
+        const savedAugmentToUnlink = savedAugmentForThisVisualization.find(
+          (savedObject) =>
+            get(savedObject, 'pluginResourceId', '') === detectorToUnlink.id
+        );
+        const savedObjectToUnlinkId = get(savedAugmentToUnlink, 'id', '');
+        await savedObjectLoader
+          .delete(savedObjectToUnlinkId)
+          .then(async (resp: any) => {
+            core.notifications.toasts.addSuccess({
+              title: `Association removed between the ${detectorToUnlink.name}
+              and the ${embeddableTitle} visualization`,
+              text: "The detector's anomalies do not appear on the visualization. Refresh your dashboard to update the visualization",
```

Comment:
2 things:
1. Should the first sentence be something like "The detector's anomalies may still appear on the visualization"? Because from the second sentence, it seems that's the case
2. Was it determined from UX that we don't want to auto-refresh for the user?

Reply:
1. This makes sense to me, I'll confirm with UX as they gave wording but it does seem like a logic error.
2. Yes currently UX is okay with not auto-refreshing, also both alerting folks and I haven't been able to get the auto refresh of the dashboard to work even if we wanted it.

Reply:
Got it.